openwisp_notifications/base/views.py

```diff
@@ -0,0 +1,44 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.http import Http404
+from django.utils.translation import gettext as _
+from django.views.generic import TemplateView
+
+User = get_user_model()
+
+
+class NotificationPreferencePage(LoginRequiredMixin, UserPassesTestMixin, TemplateView):
+    template_name = 'openwisp_notifications/preferences.html'
+    login_url = '/admin/login/'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        user_id = self.kwargs.get('pk')
+
+        if user_id:
+            try:
+                user = User.objects.get(pk=user_id)
+                # Only admin should access other users preferences
+                if not self.request.user.is_staff:
```

Comment:
Only superadmins are allowed for now.

Hence we should reject `if self.request.user.id != user_id and not user.is_superuser`.